mydata.R

```diff
@@ -23,16 +21,35 @@ get_data_fromDB<-function(credentials=credentials, sqlquery){
 
 }
 
+get_seasons <- function(df) {
+  'adds a seasons column to the data according to the south african seasons'
+  df1$season <- df1$month
+  
+  df1[df1$season=="Dec"| df1$season=="Jan"| df1$season== "Feb","season"] <- "summer"
+  df1[df1$season=="Mar"| df1$season=="Apr"| df1$season== "May","season"] <- "autumn"
+  df1[df1$season=="Jun"| df1$season=="Jul"| df1$season== "Aug","season"] <- "winter"
+  df1[df1$season=="Sep"| df1$season=="Oct"| df1$season== "Nov","season"] <- "spring"
+  
+  return(df1)
+}
+
+
+
 get_min_date <- "SELECT min(malaria.create_date_time) FROM malaria24.ona_reportedcase AS malaria"
 get_max_date <- "SELECT max(malaria.create_date_time) FROM malaria24.ona_reportedcase AS malaria"
 
-get_malaria_Data <-"SELECT 
-                    malaria.create_date_time, malaria.locality,malaria.abroad, malaria.gender, 
-                    clinic.longitude, clinic.latitude, clinic.district, clinic.subdistrict, clinic.facility
-                    FROM malaria24.ona_reportedcase AS malaria
-                    LEFT JOIN
-                    clinic_schema.clinic_database_updated_20160929 AS clinic
-                    ON malaria.facility_code::text = clinic.facilitycode::text
+
+get_malaria_Data <-"SELECT *, 
+                    EXTRACT ('month' FROM date_reported) as month
+                    FROM
+                      (SELECT 
+                       malaria.create_date_time AS date_reported,  malaria.locality,malaria.abroad, malaria.gender, 
+                       clinic.longitude, clinic.latitude, substring(clinic.district from 4) AS district, 
+                       substring(clinic.district from 1 for 2) AS province, clinic.subdistrict, clinic.facility
+                       FROM malaria24.ona_reportedcase AS malaria
+                       LEFT JOIN
+                       clinic_schema.clinic_database_updated_20160929 AS clinic
+                       ON malaria.facility_code::text = clinic.facilitycode::text) AS malaria_reports
                                  "
 
 get_time_Data <- "WITH weekly_count_table AS 
```

Comment:
I think let's add a bit more information to this name here indicating that it's a string of a SQL query. You might also want to move it above where you first use it.